File: java/remove_linked_list_elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if (head==null) return null;
        ListNode prev=null, curr=head, next=null, newHead=null;
        while (curr!=null) {
            if (curr.val==val) {
                if (prev!=null) {
                    prev.next=curr.next;
                }
            } else {
                if (newHead==null) newHead=curr;
                prev=curr;
            }
            
            curr=curr.next;
        }
        return newHead;
    }
}

File: python/remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        dummy=ListNode(0)
        dummy.next=head
        prev, curr=dummy, head
        while curr!=None:
            if curr.val==val:
                prev.next=curr.next
            else:
                prev=curr
            curr=curr.next
        
        return dummy.next
